server/mailet/rate-limiter-redis/src/main/scala/org/apache/james/rate/limiter/redis/RedisHealthCheck.scala

```diff
@@ -47,16 +49,18 @@ class RedisHealthCheck @Inject()(redisConfiguration: RedisRateLimiterConfigurati
     if (redisConfiguration.isCluster) {
       val redisUris = redisConfiguration.redisURI.value.asJava
       redisUris.forEach(redisUri => redisUri.setTimeout(healthcheckTimeout))
+      val redisClusterClient = RedisClusterClient.create(redisUris)
 
-      SMono.fromPublisher(Mono.using(() => RedisClusterClient.create(redisUris),
-        (redisClusterClient: RedisClusterClient) => Mono.fromCallable(() => redisClusterClient.connect()),
-        (redisClusterClient: RedisClusterClient) => redisClusterClient.close()))
+      SMono.fromFuture(redisClusterClient.connectAsync(StringCodec.UTF8).asScala)
+        .doOnNext(redisConnection => redisConnection.closeAsync())
+        .doOnTerminate(() => redisClusterClient.close())
```

Comment:
That one is still blocking?

Reply:
Which ones do you mean?

`redisClusterClient.connectAsync `+ `redisConnection.closeAsync` -> non blocking
`redisClusterClient.close()` -> release Java objects, should be fast enough.

Reply:
Ah actually there is a `shutdownAsync` for this (async close both connection + java objects)